File: word/src/main/java/com/dreampany/word/ui/model/DemoItem.java
package com.dreampany.word.ui.model;

import androidx.annotation.LayoutRes;
import android.view.View;

import com.dreampany.word.data.model.Demo;
import com.dreampany.frame.ui.adapter.SmartAdapter;
import com.dreampany.frame.ui.model.BaseItem;

import org.jetbrains.annotations.NotNull;

import java.io.Serializable;
import java.util.List;

import eu.davidea.flexibleadapter.FlexibleAdapter;
import eu.davidea.flexibleadapter.items.IFlexible;

/**
 * Created by Hawladar Roman on 6/29/2018.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 */
public class DemoItem extends BaseItem<Demo, DemoItem.ViewHolder> {

    private DemoItem(Demo scan, @LayoutRes int layoutId) {
        super(scan, layoutId);
    }

    public static DemoItem getSimpleItem(Demo scan, @LayoutRes int layoutId) {
        return new DemoItem(scan, layoutId);
    }

    @Override
    public ViewHolder createViewHolder(View view, FlexibleAdapter<IFlexible> adapter) {
        return new ViewHolder(view, adapter);
    }

    @Override
    public void bindViewHolder(FlexibleAdapter<IFlexible> adapter, ViewHolder holder, int position, List<Object> payloads) {
        holder.bind(this);
    }

    @Override
    public boolean filter(Serializable constraint) {
        return false;
    }

    static class ViewHolder extends BaseItem.ViewHolder {

        ViewHolder(@NotNull View view, @NotNull FlexibleAdapter adapter) {
            super(view, adapter);
        }

        void bind(DemoItem item) {
        }
    }
}

File: demo/src/main/kotlin/com/dreampany/demo/ui/model/DemoItem.kt
package com.dreampany.demo.ui.model

import android.view.View
import androidx.annotation.LayoutRes
import androidx.recyclerview.widget.RecyclerView
import com.dreampany.demo.data.model.Demo
import com.dreampany.demo.ui.adapter.DemoAdapter
import com.dreampany.frame.data.model.BaseKt
import com.dreampany.frame.ui.model.BaseItemKt
import eu.davidea.flexibleadapter.FlexibleAdapter
import eu.davidea.flexibleadapter.items.IFlexible
import java.io.Serializable

/**
 * Created by Roman-372 on 7/24/2019
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
class DemoItem private constructor(item: Demo, @LayoutRes layoutId: Int = 0) :
    BaseItemKt<Demo, DemoItem.ViewHolder, String>(item, layoutId) {

    companion object {
        fun getItem(item: Demo): DemoItem {
            return DemoItem(item)
        }
    }

    override fun createViewHolder(
        view: View,
        adapter: FlexibleAdapter<IFlexible<RecyclerView.ViewHolder>>
    ): ViewHolder {
        return ViewHolder(view, adapter)
    }

    override fun filter(constraint: String): Boolean {
        return item.id.startsWith(constraint, true);
    }

    class ViewHolder(view: View, adapter: FlexibleAdapter<*>) :
        BaseItemKt.ViewHolder(view, adapter) {

        private var adapter: DemoAdapter

        init {
            this.adapter = adapter as DemoAdapter
        }

        override fun <VH : BaseItemKt.ViewHolder, T : BaseKt, S : Serializable, I : BaseItemKt<T, VH, S>>
                bind(position: Int, item: I) {
            val uiItem = item as DemoItem
            val word = uiItem.item
        }
    }
}